src/ore/src/tracing.rs

```diff
@@ -489,51 +489,52 @@ impl OpenTelemetryContext {
     /// to create a context, then the current thread's `Context` is used
     /// defaulting to the default `Context`.
     pub fn attach_as_parent(&self) {
-        let parent_cx = global::get_text_map_propagator(|prop| prop.extract(&self.inner));
+        let parent_cx = global::get_text_map_propagator(|prop| prop.extract(self));
         tracing::Span::current().set_parent(parent_cx);
     }
 
     /// Obtains a `Context` from the current [`tracing`] span.
     pub fn obtain() -> Self {
-        let mut map = std::collections::HashMap::new();
+        let mut context = Self::empty();
         global::get_text_map_propagator(|propagator| {
-            propagator.inject_context(&tracing::Span::current().context(), &mut map)
+            propagator.inject_context(&tracing::Span::current().context(), &mut context)
         });
 
-        Self { inner: map }
+        context
     }
 
     /// Obtains an empty `Context`.
     pub fn empty() -> Self {
         Self {
-            inner: HashMap::new(),
+            inner: BTreeMap::new(),
         }
     }
 }
 
 impl Extractor for OpenTelemetryContext {
     fn get(&self, key: &str) -> Option<&str> {
-        Extractor::get(&self.inner, key)
+        self.inner.get(&key.to_lowercase()).map(|v| v.as_str())
     }
+
     fn keys(&self) -> Vec<&str> {
-        Extractor::keys(&self.inner)
+        self.inner.keys().map(|k| k.as_str()).collect::<Vec<_>>()
     }
 }
 
 impl Injector for OpenTelemetryContext {
     fn set(&mut self, key: &str, value: String) {
-        Injector::set(&mut self.inner, key, value)
+        self.inner.insert(key.to_lowercase(), value);
     }
 }
```

Comment:
There is no `Extractor`/`Injector` impl for `BTreeMap` we could referred to. So instead I copied the logic from the `HashMap` impl here.